sherpa/ui/utils.py

```diff
@@ -17072,15 +17410,16 @@ def get_model_image(self, id=None):
 
     # DOC-TODO: it looks like get_source_image doesn't raise DataErr with
     # a non-2D data set
-    def get_source_image(self, id=None):
+    def get_source_image(self,
+                         id: Optional[IdType] = None):
         """Return the data used by image_source.
 
         Evaluate the source expression for the image pixels - without
         any PSF convolution - and return the results.
 
         Parameters
         ----------
-        id : int or str, optional
+        id : int, str, or None. optional
```

Comment:
```suggestion
        id : int, str, or None, optional
```

Reply:
argh argh

Or should that be $\rm{argh}^2$